azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -176,6 +176,247 @@ func resourceArmEventGridEventSubscription() *schema.Resource {
 				},
 			},
 
+			"advanced_filter": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bool_equals": {
+							Type:     schema.TypeList,
```

Comment:
this (along with the other nested blocks) may want changing to a Set depending on how the API behaves with multiple elements, but this is fine for now 👍

Reply:
It seems that duplicate values are allowed (tested by myself). But in fact a set would make more sense.

Reply:
Unfortunately testing is a lot more complex using sets, as one has to know the computed hash value of the properties to test their values. Do you have an idea how to easily solve this?